plugins/destination/snowflake/client/write.go

```diff
@@ -54,13 +55,20 @@ func (c *Client) WriteTableBatch(ctx context.Context, name string, msgs message.
 	if err := f.Close(); err != nil {
 		return fmt.Errorf("failed to close temp file with last resource %s: %w", f.Name(), err)
 	}
-	sql := fmt.Sprintf(putFileIntoStage, f.Name())
+
+	sql := fmt.Sprintf(putFileIntoStage, escapePath(f.Name()))
+
 	if _, err := c.db.ExecContext(ctx, sql); err != nil {
 		return fmt.Errorf("failed to put file into stage with last resource %s: %w", sql, err)
 	}
-	sql = fmt.Sprintf(copyIntoTable, tableName, path.Base(f.Name()))
+	sql = fmt.Sprintf(copyIntoTable, tableName, escapePath(path.Base(f.Name())))
 	if _, err := c.db.ExecContext(ctx, sql); err != nil {
 		return fmt.Errorf("failed to copy file into table with last resource %s: %w", sql, err)
 	}
 	return err
 }
+
+// escapePath properly escapes the `\` character in window's file paths.
+func escapePath(p string) string {
+	return strings.ReplaceAll(p, "\\", "\\\\")
```

Comment:
Is escaping the backslash the same as converting to a forward slash - which the docs recommended?

Reply:
The solution actually comes from the Snowflake driver docs:
https://pkg.go.dev/github.com/snowflakedb/gosnowflake#hdr-Support_For_PUT_and_GET